src/main/java/controller/LadderGameController.java

```diff
@@ -0,0 +1,112 @@
+package controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import domain.ladder.LadderGame;
+import domain.ladder.LadderHeight;
+import domain.player.Player;
+import domain.player.Players;
+import domain.prize.Prizes;
+import view.InputView;
+import view.OutputView;
+
+public class LadderGameController {
+
+	private final InputView inputView;
+	private final OutputView outputView;
+	private final LadderGame ladderGame;
+
+	public LadderGameController(InputView inputView, OutputView outputView, LadderGame ladderGame) {
+		this.inputView = inputView;
+		this.outputView = outputView;
+		this.ladderGame = ladderGame;
+	}
+
+	public void run() {
+		Players players = retryOnException(this::getPlayers);
+		int playerCount = players.getPlayerCount();
+
+		Prizes prizes = retryOnException(() -> getPrizes(playerCount));
+		LadderHeight height = retryOnException(this::getHeight);
+
+		ladderGame.createRandomLadder(playerCount, height);
+
+		printCurrentLadder(players, prizes);
+		printPlayerResult(players, prizes);
+	}
+
+	private Players getPlayers() {
+		outputView.printReadNames();
+		List<String> names = inputView.readNames();
+
+		List<Player> players = new ArrayList<>();
+		for (int order = 0; order < names.size(); order++) {
```

Comment:
```suggestion
		for (int i = 0; i < names.size(); i++) {
```
i를 이용해주세요 오히려 혼란을 주는것 같아요
그리고 stream을 활용해보면 어떨까요?

Reply:
> i를 이용해주세요 오히려 혼란을 주는것 같아요 그리고 stream을 활용해보면 어떨까요?

저도 가급적 반복은 스트림으로 처리하고자 하는데, 이렇게 인덱스값을 직접 이용해야하는 경우는 for문을 사용하는 것이 더 직관적으로 느껴진다는 생각이었습니다. 

```java
List<Player> players = IntStream.range(0, names.size())
      .map(i -> new Player(names.get(i), i))
      .toList
```

스트림으로 구현하면 위와 같은 방법으로 해야할 것 같은데, IntStream을 이용하는 것이 그렇게 보기 좋다는 생각이 들지 않더라구요. 혹여나 다른 의견이 있으시면 부탁드립니다!